include/CL/opencl.hpp

```diff
@@ -10463,6 +10505,28 @@ class Semaphore : public detail::Wrapper<cl_semaphore_khr>
         return param;      
     }
 
+#ifdef cl_khr_external_semaphore
+    cl_int getSemaphoreHandleForTypeKHR (Device device,
+        ExtSemaphoreHandleTypeKhr handle_type,
+        size_type handle_size,
+        void* handle_ptr,
+        size_type* handle_size_ret)
```

Comment:
This is OK, but I think I'd really want to write code more like this:

```c++
auto my_handle = my_semaphore->getHandleForTypeKHR<TypeD3D12Fence>(Device);
```

Note, this is similar to other queries, like the different `getInfo`, `getWorkGroupInfo`, and `getSubGroupInfo` queries.

What do you think?

Reply:
For completeness, I think we could also have a variant that takes a parameter for the handle type, for cases where the handle type is only known at runtime, but I still think we'd want it to be more like:

```c++
cl_int getHandleForType<T>(const Device& device, ExtSemaphoreHandleTypeKhr handle_type, T* handle_ptr);
```

This would let you write code like this that doesn't need to worry about `void*`s.

```c++
FILE fp;
my_semaphore->getHandleForTypeKHR(my_device, TypeOpaqueFd, &fp);
```

Note, there is a variant like this for `getInfo`, also.

Reply:
I think it has a sense :)
Done, thanks.

Reply:
Thanks, but this isn't quite what I was looking for.  Have a look at: https://github.com/Oskar-Plaska-Mobica/OpenCL-CLHPP/pull/1